### PR TITLE
Add nf-test sharding CI

### PR DESCRIPTION
**Pull request by edmundmiller**
- **ci: Copy over from methylseq**
- **ci: Move Sarek specific settings to ci**
- **ci: tap => xml**

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by maxulysse** [comment]
I guess now you'd have a couple of merge conflicts

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -22,303 +19,54 @@ env:
...
-    runs-on: ubuntu-latest
-    needs: pytest-changes
-    if: needs.pytest-changes.outputs.tags != '[]'
+    name: ${{ matrix.NXF_VER }} | ${{ matrix.filter }} | ${{ matrix.profile }} | ${{ matrix.shard }}
     strategy:
       fail-fast: false
       matrix:
-        tags: ["${{ fromJson(needs.pytest-changes.outputs.tags) }}"]
+        # BUG: "latest-everything" https://github.com/nf-core/sarek/issues/1672
+        NXF_VER: ["23.04.0"] # TODO Test minimum version on release
```
Comment (maxulysse) [inline .github/workflows/ci.yml:34]:
```suggestion
        NXF_VER:
          - "23.04.0"
          - "latest-everything"
```

File: .github/nf-test-tags.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  - tags: "strelka/somatic"
+  - tags: "subworkflows/utils_nfvalidation_plugin"
+  - tags: "tabix/bgziptabix"
+  - tags: "tabix/tabix"
+  - tags: "tiddit/sv"
+  - tags: "untar"
+  - tags: "pipeline_sarek"
+include:
+  - tags: "pipeline_sarek"
+    profile: "test,docker"
```
Comment (maxulysse) [inline .github/nf-test-tags.yml:54]:
```suggestion
```

**Review by maxulysse — DISMISSED** [review]
Works as expected, thanks @edmundmiller

**Comment by maxulysse** [comment]
How to exclude the tags you moved in `nf-test-tags.yml`?

**Comment by edmundmiller** [comment]
> How to exclude the tags you moved in `nf-test-tags.yml`?

That's a tough one. I've got https://github.com/askimed/nf-test/issues/260 which we could add to the conda runs. 

We could add a `conda_pass` to everything that works, but that sounds painful.
I wonder if we could add a conditional assert if the the profile is conda in the test itself?

Anything that just fails no matter the container abstraction, just mark it as failing with a note that it shouldn't be failing, but ironing out every test is a daunting task that we should handle in a separate PR.

**Comment by maxulysse** [comment]
merged in https://github.com/nf-core/sarek/pull/1678
Looking good

**Comment by maxulysse** [comment]
Synced with dev to include TEMPLATE merge

**Comment by maxulysse** [comment]
the last 2 failing tests are fixed in https://github.com/nf-core/sarek/pull/1678

**Review by FriederikeHanssen — APPROVED** [review]

**Review by FriederikeHanssen — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -22,288 +20,65 @@ env:
...
       fail-fast: false
       matrix:
-        tags: ["${{ fromJson(needs.pytest-changes.outputs.tags) }}"]
-        profile: ["docker"]
-        # profile: ["docker", "singularity", "conda"]
-        TEST_DATA_BASE:
-          - "test-datasets/data"
         NXF_VER:
           - "24.04.2"
           - "latest-everything"
```
Comment (maxulysse) [inline .github/workflows/ci.yml:36]:
```suggestion
          # - "latest-everything"
```

**Pull request merged**